DatetimeIndex comparison with ndim>1 incorrect or raises

Discovered while testing #18162.

```
dti = pd.date_range('2016-01-01', periods=3, tz='US/Pacific')
df = pd.Series(dti).to_frame()
arr = np.array([dti.tolist()]).T  # <-- object dtype, correctly retains TZ
arr2 = np.array([dti]).T          # <-- datetime64[ns], drops TZ

>>> dti == df
[...]
  File "pandas/core/indexes/datetimes.py", line 145, in _ensure_datetime64
    raise TypeError('%s type object %s' % (type(other), str(other)))

>>> df == dti
  File "<stdin>", line 1, in <module>
  File "pandas/core/ops.py", line 1302, in f
    try_cast=False)
  File "pandas/core/frame.py", line 4012, in _combine_const
[...]
  File "pandas/core/internals.py", line 121, in __init__
    '{mgr}'.format(val=len(self.values), mgr=len(self.mgr_locs)))
ValueError: Wrong number of items passed 3, placement implies 1

>>> dti == arr
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pandas/core/indexes/datetimes.py", line 126, in wrapper
    o_mask = other.view('i8') == libts.iNaT
  File "/usr/local/lib/python2.7/site-packages/numpy/core/_internal.py", line 377, in _view_is_safe
    raise TypeError("Cannot change data-type for object array.")
TypeError: Cannot change data-type for object array.

>>> dti == arr2    # <-- this one is fixed by #18162
array([[ True, False, False],
       [False,  True, False],
       [False, False,  True]], dtype=bool)

>>> df == arr   # <-- this one is OK
      0
0  True
1  True
2  True

>>> df == arr2  # <-- should raise TypeError
       0
0  False
1  False
2  False

>>> df[0] == arr2[:, 0]  # <-- should raise TypeError
0    True
1    True
2    True
```

Seems like `df[0] == arr2[:, 0]` fails an equality check. Looks like one is tz-aware the other is not.
What would you like to do here @mroeschke? 

last line of my test: 
```python
   def test_date_range_index_comparison(self):
        rng = date_range("2011-01-01", periods=3, tz="US/Eastern")
        df = Series(rng).to_frame()
        arr = np.array([rng.to_list()]).T
        arr2 = np.array([rng]).T

        with pytest.raises(ValueError):
            rng == df

        with pytest.raises(ValueError):
            df == rng

        tm.assert_numpy_array_equal(df.values, arr)
        tm.assert_numpy_array_equal(df.values, arr2)
        tm.assert_numpy_array_equal(df[0].values, arr2[:, 0])
```

Results in: 
```python
E           AssertionError: numpy array are different
E           
E           numpy array values are different (100.0 %)
E           [left]:  [2011-01-01T05:00:00.000000000, 2011-01-02T05:00:00.000000000, 2011-01-03T05:00:00.000000000]
E           [right]: [2011-01-01 00:00:00-05:00, 2011-01-02 00:00:00-05:00, 2011-01-03 00:00:00-05:00]```

--- a/pandas/tests/indexes/datetimes/test_date_range.py
+++ b/pandas/tests/indexes/datetimes/test_date_range.py
@@ -26,7 +26,9 @@ import pandas.util._test_decorators as td
 
 import pandas as pd
 from pandas import (
+    DataFrame,
     DatetimeIndex,
+    Series,
     Timedelta,
     Timestamp,
     bdate_range,
@@ -354,6 +356,34 @@ class TestDateRanges:
             ]
         )
         tm.assert_index_equal(result, expected)
+
+    def test_date_range_index_comparison(self):
+        rng = date_range("2011-01-01", periods=3, tz="US/Eastern")
+        df = Series(rng).to_frame()
+        arr = np.array([rng.to_list()]).T
+        arr2 = np.array([rng]).T
+
+        with pytest.raises(ValueError, match="Unable to coerce to Series"):
+            rng == df
+
+        with pytest.raises(ValueError, match="Unable to coerce to Series"):
+            df == rng
+
+        expected = DataFrame([True, True, True])
+
+        results = df == arr2
+        tm.assert_frame_equal(results, expected)
+
+        expected = Series([True, True, True], name=0)
+
+        results = df[0] == arr2[:, 0]
+        tm.assert_series_equal(results, expected)
+
+        expected = np.array(
+            [[True, False, False], [False, True, False], [False, False, True]]
+        )
+        results = rng == arr
+        tm.assert_numpy_array_equal(results, expected)
 
     @pytest.mark.parametrize(
         "start,end,result_tz",

